### PR TITLE
Add k8s 1.17 release notes to 4.2.0

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
Signed-off-by: Jordi Massaguer Pla <jmassaguerpla@suse.de>

### CODE REVIEW

**Review by evrardjp — DISMISSED** [review]
I am not sure we need L28 to 36, as these are not impacting deployers/end-users per se. The "removals" here don't mean removal of the feature.

I would keep the link to the CHANGELOG though. 

I think I linked you in another PR about the changes in k8s. I would add those into this PR.

**Comment by evrardjp** [comment]
> I think I linked you in another PR about the changes in k8s. I would add those into this PR.

This is what I would add: https://github.com/kubernetes/kubernetes/pull/85903

**Comment by jordimassaguerpla** [comment]
@nkoranova or @r0ckarong please review. thanks.

File: adoc/MAIN.release-notes.adoc [hunk 1]
```diff
@@ -21,20 +21,34 @@ To make such entries easier to identify, they contain a note to that effect.
 
 == Changes in 4.2.0
 
+=== Deprecations in 4.2.0
```
Comment (r0ckarong) [inline adoc/MAIN.release-notes.adoc:24]: @jordimassaguerpla Do all of these really apply to 4.2.0? The section heading was previously 4.2.x but that doesn't make sense in this document.

Comment (jordimassaguerpla) [reply adoc/MAIN.release-notes.adoc:24]: yes, they apply to 4.2.0

**Comment by jordimassaguerpla** [comment]
qa gave thumbsup. We can proceed with this.

**Pull request merged**